### PR TITLE
Blackstone Port - Custom Locks (+ Fix Update)

### PR DESCRIPTION
**Pull request by Stutternov**
## About The Pull Request

Tl;dr - Ports both the ability to make custom keys and locks from Blackstone; plus the fix PR they merged yesterday at the time of writing.
https://github.com/Blackstone-SS13/BLACKSTONE/pull/680
https://github.com/Blackstone-SS13/BLACKSTONE/pull/1154

**I'm personally not sure how I feel about this feature, but if Blackstone hasn't had issues we shouldn't I guess.** I think the idea of player-built homes is a really cool feature, but due to other servers with similar mechanics I've hated the feature to an extent as well. (I.e - person building a door in front of someone else to fuck them over like we're playing Rust and you have TC access.....)

Might be worth testmerging first; this was a heavily requested feature people said Blackstone had, and the person who put this code together over there did so pretty well. So - hey, worth a shot.

## Why It's Good For The Game

Heavily requested feature. Adds flavor to the game + gives blacksmiths something a lot of people will probably want crafted.

Downside: We will see the start of fuck-shacks..............

## Pre-Merge Checklist

- [ ] You tested this on a local server.
- [ ] This code did not runtime during testing.
- [X] You documented all of your changes.

### CODE REVIEW

**Comment by messagecypher** [comment]
I think this fucked the keys, as wooden doors cannot be unlocked with their usual keys

**Comment by Stutternov** [comment]
There, should be fixed now. Just - re-added wood doors being default locked; but no idea if that will fuck with new doors crafted currently. Will have to test later.
